packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2030,16 +2446,24 @@ export class TransactionController extends BaseControllerV1<
         return;
       }
 
+      let { nonceTracker } = this;
+      if (networkClientId) {
+        const trackers = this.trackingMap.get(networkClientId);
```

Comment:
This is good example of why a `MultichainHelper` could be helpful as we could encapsulate logic like this into a `getNonceTracker(networkClientId: NetworkClientId)` that could `throw` and return just the `NonceTracker`?